Repairing the integrated build path to handle the Caffe2 PR.

Jorghi12 opened the pull request:
This PR handles the recent Caffe2 PR that modified the build path last night.

Jorghi12 committed:
Modifying the build path to handle Caffe2's merge

bddppq commented on cmake/public/LoadHIP.cmake:
```diff
@@ -126,6 +133,11 @@ IF(HIP_FOUND)
 
+
+  # Necessary includes for building PyTorch since we include HIP headers that depend on hcc/hsa headers.
+  set(hsa_INCLUDE_DIRS ${HCC_PATH}/include)
```
can this be found via `find_package(hsa REQUIRED)` and let their cmake files populate the has_INCLUDE_DIRS?
line 109 - 119

bddppq commented on cmake/Dependencies.cmake:
```diff
@@ -459,9 +459,7 @@ if(BUILD_CAFFE2 OR BUILD_ATEN)
     set(Caffe2_HIP_INCLUDES
-      ${hip_INCLUDE_DIRS} ${rocrand_INCLUDE_DIRS} ${hiprand_INCLUDE_DIRS} ${rocblas_INCLUDE_DIRS} ${miopen_INCLUDE_DIRS} ${thrust_INCLUDE_DIRS} $<INSTALL_INTERFACE:include> ${Caffe2_HIP_INCLUDES})
-    # This is needed for library added by hip_add_library (same for hip_add_executable)
-    hip_include_directories(${Caffe2_HIP_INCLUDES})
```
this is better to leave here as it applies to all libraries/executable added via hip_add_library/hip_add_executable. 
Caffe2 has two hip_add_library (libcaffe2_hip.so and the python binding) and several hip_add_executable (c++ tests)

bddppq replied:
I see. In that case let's just move it from Dependencies to caffe2/CMake as what you have done initially (add a note to prevent other people to move it back again :-))

Commit:
Update LoadHIP.cmake

Fixing typo.

Commit:
Update Dependencies.cmake

Keeping hip_include_directories since other Caffe2 libs depend on it.

Commit:
Update CMakeLists.txt

Only including for the second time if we're building with ATen.

ezyang commented:
@Jorghi12 If you know why you are making the changes to the cmake, take the time to go over the diff and write down in English why particular changes were made. It helps a lot when people from the future are spelunking over the diff.

Commit:
Update CMakeLists.txt

Adding comments to make sure future users understand why necessary commands have been added.
